app/src/main/java/com/portfolio/tasky/usecases/EmailPatternValidatorImpl.kt

```diff
@@ -0,0 +1,12 @@
+package com.portfolio.tasky.usecases
+
+import android.util.Patterns
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class EmailPatternValidatorImpl @Inject constructor() : EmailPatternValidator {
+    override fun isValidEmailPattern(email: CharSequence?): Boolean {
+        return Patterns.EMAIL_ADDRESS.matcher(email as CharSequence).matches()
+    }
```

Comment:
This implementation should be in the "data" portion of the feature

Check the video about "Proper Package Structure" (or however it's called) to know more about how to structure the app

Reply:
Sure I ill look at the structure video again but if it's ok then we can skip this one for now. I'll rectify the package structure in the next PR since it will require only refactoring of file locations.